lib/src/rules/directives_ordering.dart

```diff
@@ -229,6 +302,31 @@ class _Visitor extends SimpleAstVisitor {
         .forEach(reportDirective);
   }
 
+  void _checkSectionInOrder(
+      Set<AstNode> lintedNodes, Iterable<NamespaceDirective> nodes) {
+    void reportDirective(NamespaceDirective directive) {
+      if (lintedNodes.add(directive)) {
+        rule._reportLintWithDirectiveSectionOrderedAlphabeticallyMessage(
+            directive);
+      }
+    }
+
+    List<NamespaceDirective> addIfUpperBound(
+        List<NamespaceDirective> acc, NamespaceDirective curr) {
+      if (acc.isEmpty || acc.last.uriContent.compareTo(curr.uriContent) < 0) {
+        acc.add(curr);
+      }
+
+      return acc;
+    }
+
+    final inOrderNodes = nodes.fold(<NamespaceDirective>[], addIfUpperBound);
+
+    nodes.toSet()
+      ..removeAll(inOrderNodes)
+      ..forEach(reportDirective);
```

Comment:
It would be more efficient to reverse the selection choice in `addIfUpperBound` (so that it adds directives that are not in order). That way you wouldn't need to convert `nodes` to a set, nor remove items from the set.